src/main/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlight.java

```diff
@@ -292,6 +295,16 @@ public SnapshotCreateFlight(FlightMap inputParameters, Object applicationContext
     // The IAM code contains retries, so we don't make a retry rule here.
     addStep(new SnapshotAuthzIamStep(iamClient, snapshotService, snapshotReq, userReq, snapshotId));
 
+    // add data access control groups to the snapshot
+    List<String> dataAccessControlGroups = snapshotReq.getDataAccessControlGroups();
+    if (Objects.nonNull(dataAccessControlGroups) && !dataAccessControlGroups.isEmpty()) {
+      addStep(
+          new CreateSnapshotGroupConstraintPolicyStep(
+              policyService, snapshotId, dataAccessControlGroups));
+      addStep(
+          new AddSnapshotAuthDomainStep(iamService, userReq, snapshotId, dataAccessControlGroups));
```

Comment:
I got it wrong in Slack -- It's actually adding an auth domain in Sam that cannot be undone.

https://github.com/DataBiosphere/jade-data-repo/blob/bd4cbaef6265ca000521bfadc692d8615d0dadab/src/main/java/bio/terra/service/snapshot/flight/authDomain/AddSnapshotAuthDomainStep.java#L17

This poses some concern when adding DACs to existing snapshots: if that flight fails partway and has to walk back, the snapshot wouldn't have its auth domain removed.  But, I think this is not a problem when we add DACs to snapshots on **creation** -- if the flight ends up failing and walks back, the whole Sam resource is going to be deleted anyway, so it doesn't matter that we can't remove its auth domain.

Reply:
> This poses some concern when adding DACs to existing snapshots